src/main/java/gov/cabinetofice/gapuserservice/service/OneLoginService.java

```diff
@@ -141,4 +151,27 @@ public Optional<User> getUser(final String email, final String sub) {
         if (userBySub.isPresent()) return userBySub;
         return userRepository.findByEmail(email);
     }
+
+    public String generateNonce() {
+         nonce = UUID.randomUUID().toString();
```

Comment:
Can just return `UUID.randomUUID().toString();` directly for now

We can validate state and nonce in a different ticket - i expect we will store them in the user service cookie rather than in class